### ISSUE TITLE
Cannot set boolean property from command line using --set

### ISSUE THREAD

**Issue by mfabiani-av** [Azure/aks-engine#1778]
**Describe the bug**
I'm trying to set orchestratorProfile.kubernetesConfig.privateCluster.enabled=true from the command line using --set but I get the following error:

`INFO[0000] new api model file has been generated during merge: /tmp/mergedApiModel298698370
Error: loading API model: error parsing the api model: json: cannot unmarshal string into Go struct field Properties.orchestratorProfile of type bool`

**Steps To Reproduce**
Run something like this:
`aks-engine deploy --api-model ./cfg/autoscaler_jump.json --set orchestratorProfile.kubernetesConfig.privateCluster.enabled=true --location uksouth`

**Expected behavior**
Override value in my API config file

**AKS Engine version**
0.39.1

**Kubernetes version**

**Additional context**

**Comment by adelina-t** [comment]
Yeah, I've been seeing this as well in some of my test scripts. Fix should be pretty straightforward.
